example: Update `example_cli` and retire old nursery crates

ValuedMammal opened the pull request:
- Adds two commands `init` and `generate`. Loading database doesn't require descriptors
- Replaces `send` command with `psbt` (new, sign, and extract). Supports tap key spend
- Uses `bdk-coin-select` and miniscript `plan` module

fixes #1469 
partially addresses #1462

### Notes to the reviewers

Note the `example_cli` lib now defines the `ChangeSet` and `Anchor` type whereas before these were generic

### Changelog notice

* Improvements to `example_cli` that include generating descriptors and creating PSBTs

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

ValuedMammal committed:
example: Update example_cli

storopoli commented:
We could merge this with #1442 and then commit it as a single PR?

If favorable I could rebase #1442 and then you can commit on top of that? Or I can do the opposite as well...

ValuedMammal commented:
1442 can be merged anytime and from what I can tell doesn't conflict with this one

rustaceanrob commented:
Nice. This cleans things up quite a bit.

ValuedMammal commented:
* Rebased onto 17a9850cba06db837b82cef1dbaf2ef1e342f7ec
* For now I'm using `bdk_wallet::ChangeSet` for the example-cli
* Added back support for single descriptor use
* CI failed to build msrv

LLFourn commented on example-crates/example_cli/Cargo.toml:
```diff
@@ -7,11 +7,11 @@ edition = "2021"
 
 [dependencies]
 bdk_chain = { path = "../../crates/chain", features = ["serde", "miniscript"]}
+bdk_wallet = { path = "../../crates/wallet" }
```
Can we not depend on `bdk_wallet` for its changeset. Just redefine it here for now. I know it's friction and seems like a waste of time but the point of the `example_cli` examples is more to demonstrate to us as developers what you have to do to make a wallet with `bdk_chain` + the backends. There is of course still way too much you have to do but it's better not to hide it by depending on `bdk_wallet`.

LLFourn reviewed (commented):
ConceptACK this looks great.

evanlinjin reviewed (approved):
ACK 771f6b9c97a70be2e8d2da5216f3b686295861bd

A lot better than what we had before. Made the examples a lot more useful.

ValuedMammal commented:
Thanks @evanlinjin @LLFourn 

We're currently limited to creating PSBTs for `wpkh` and `tr` descriptors because I got stuck on trying to add assets/keys for `wsh` (can still sync and derive addresses though).

evanlinjin commented:
@ValuedMammal I think that's fine for now, can add them later
